llpc/translator/lib/SPIRV/SPIRVToLLVMDbgTran.cpp

```diff
@@ -104,16 +105,33 @@ const std::string &SPIRVToLLVMDbgTran::getString(const SPIRVId Id) {
   return String->getStr();
 }
 
-void SPIRVToLLVMDbgTran::transDbgInfo(SPIRVValue *SV, Value *V) {
+unsigned SPIRVToLLVMDbgTran::getConstant(const SPIRVId id) {
+  return BM->get<SPIRVConstant>(id)->getZExtIntValue();
+}
+
+void SPIRVToLLVMDbgTran::recordsDbgInfo(SPIRVValue *SV, Value *V) {
   if (!Enable || !SV->hasLine())
     return;
   // A constant sampler does not have a corresponding SPIRVInstruction.
   if (SV->getOpCode() == OpConstantSampler)
     return;
 
-  if (Instruction *I = dyn_cast<Instruction>(V)) {
+  if (dyn_cast<Instruction>(V) != nullptr) {
     SPIRVInstruction *SI = static_cast<SPIRVInstruction *>(SV);
-    I->setDebugLoc(transDebugScope(SI, I));
+    if (ValueMap.find(SI) == ValueMap.end()) {
+      ValueMap[SI] = nullptr;
+    }
+  }
+}
+
+void SPIRVToLLVMDbgTran::transDbgInfo() {
+  for (auto it = ValueMap.begin(); it != ValueMap.end(); ++it) {
+    if (it->second == nullptr) {
+      auto lv = SPIRVReader->getTranslatedValue(it->first);
+      SPIRVInstruction *sinst = static_cast<SPIRVInstruction *>(it->first);
+      Instruction *inst = dyn_cast<Instruction>(lv);
+      inst->setDebugLoc(transDebugScope(sinst, inst));
```

Comment:
Use `cast` instead of `dyn_cast` when you're certain about the type. This is better because it raises an assertion if the type turns out to be wrong.